tutorials/dataframe/df022_createKahan.C

```diff
@@ -0,0 +1,98 @@
+/// \file
+/// \ingroup tutorial_dataframe
+/// \notebook
+/// This tutorial shows how to implement a Kahan summation custom action.
+///
+/// \macro_code
+///
+/// \date July 2018
+/// \author Enrico Guiraud, Danilo Piparo, Massimo Tumolo
+
+class KahanSum : public ROOT::Detail::RDF::RActionImpl<class KahanSum> {
+public:
+   /// This type is a requirement for every helper.
+   using Result_t = double;
+
+private:
+   std::vector<double> fPartialSums;
+   std::vector<double> fCompensations;
+   int fNSlots;
+
+   std::shared_ptr<double> fResultSum;
+
+public:
+   KahanSum(KahanSum &&) = default;
+   KahanSum(const KahanSum &) = delete;
+
+   KahanSum(const std::shared_ptr<double> &r) : fResultSum(r)
+   {
+      fNSlots = ROOT::IsImplicitMTEnabled() ? ROOT::GetImplicitMTPoolSize() : 1;
+      fPartialSums.resize(fNSlots, 0.0);
+      fCompensations.resize(fNSlots, 0.0);
+   }
+
+   std::shared_ptr<Result_t> GetResultPtr() const { return fResultSum; }
+
+   void Initialize() {}
+   void InitTask(TTreeReader *, unsigned int) {}
+
+   template <typename T>
+   void Exec(unsigned int slot, T x)
+   {
+      double y = x - fCompensations[slot];
+      double t = fPartialSums[slot] + y;
+      fCompensations[slot] = (t - fPartialSums[slot]) - y;
+      fPartialSums[slot] = t;
+   }
+
+   template <typename T, typename std::enable_if<ROOT::TypeTraits::IsContainer<T>::value, int>::type = 0>
+   void Exec(unsigned int slot, const T &vs)
+   {
+      for (auto &&v : vs) {
+         Exec(slot, v);
+      }
+   }
+
+   void Finalize()
+   {
+      double sum = 0;
+      double compensation = 0;
+      for (int i = 0; i < fNSlots; ++i) {
+         double y = fPartialSums[i] - compensation;
+         double t = sum + y;
+         compensation = (t - sum) - y;
+         sum = t;
+      }
+      *fResultSum = sum;
+   }
+};
+
+void df022_createKahan()
+{
+   // We enable implicit parallelism
+   ROOT::EnableImplicitMT(8);
+
+   std::vector<double> source(10);
+   double standardSum = 0;
+   double ele = 0;
+   for (int i = 1; i <= 10; ++i) {
+      ele = (i % 2 == 0) ? ((double)1) / 100000000 : ((double)1) * 100000000;
```

Comment:
Could maybe `((double)1) / 100000000 : ((double)1) * 100000000` be something like 00000000.1 : 100000000;